src/daoreg.cpp

```diff
@@ -10,103 +10,170 @@ ACTION daoreg::reset() {
   while (daoit != _dao.end()) {
     daoit = _dao.erase(daoit);
   }
-
 }
 
 ACTION daoreg::create(const name& dao, const name& creator, const std::string& ipfs) {
 
   require_auth(creator);
-
+  
   dao_table _dao(get_self(), get_self().value);
   _dao.emplace(get_self(), [&](auto& new_org){
+    uint64_t dao_id = _dao.available_primary_key();
+    new_org.dao_id = dao_id;
     new_org.dao = dao;
     new_org.creator = creator;
     new_org.ipfs = ipfs;
   });
 
-  uint64_t ram_bytes = config_get_uint64(name("b.rambytes"));
-
-  action (
-         permission_level( get_self(), name("active") ),
-         name("eosio"), 
-         name("buyrambytes"),
-         std::make_tuple(get_self(), dao, uint32_t(ram_bytes))
-  ).send();
-
-  asset del_amount_net = config_get_asset(name("d.net"));
-  asset del_amount_cpu = config_get_asset(name("d.cpu"));
-
-  action(
-        permission_level(get_self(), name("active")),
-        name("eosio"),
-        name("delegatebw"),
-        std::make_tuple(get_self(), dao, del_amount_net, del_amount_cpu, true)
-  ).send();
-  
+  if (is_account(creator)) {
```

Comment:
the method `is_account` has to check for the `dao` account instead of the creator. The creator  is an account otherwise the the require_auth would've failed by this point, but we want to me optional the creation of the dao account.